src/main/java/com/devathon/preguntonic/controllers/GameControllerImpl.java

```diff
@@ -0,0 +1,279 @@
+package com.devathon.preguntonic.controllers;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import com.devathon.preguntonic.constants.Constants;
+import com.devathon.preguntonic.domain.Question;
+import com.devathon.preguntonic.dto.GameStatusDto;
+import com.devathon.preguntonic.dto.PlayerQuestionResponseDto;
+import com.devathon.preguntonic.dto.PlayerStatusDto;
+import com.devathon.preguntonic.dto.QuestionDto;
+import com.devathon.preguntonic.dto.QuestionResultDto;
+import com.devathon.preguntonic.model.Game;
+import com.devathon.preguntonic.model.GameStatus;
+import com.devathon.preguntonic.model.Player;
+import com.devathon.preguntonic.model.PlayerStatus;
+import com.devathon.preguntonic.model.Room;
+import com.devathon.preguntonic.model.RoomStatus;
+import com.devathon.preguntonic.services.DummyRoomService;
+import com.devathon.preguntonic.services.QuestionService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+public class GameControllerImpl implements GameController {
```

Comment:
NIT: if you have time we can handle a layered architecture where the controller calls a service layer.

Thus, the controller's only responsibility would be to manage the request and parameters that arrive and coordinate/orchestrate the calls to the services.

So for now the proposed architecture is:
Controller(handle api calls)-->Services(Room/Game logic)-->Repositories(database access)

WDYT?

Reply:
It is a fairly good idea, but, what logic will be in the controller and what in the service? (null checks, logic, etc.)